src/write.c

```diff
@@ -2591,21 +2591,34 @@ static avifResult avifEncoderWriteMiniBox(avifEncoder * encoder, avifRWStream *
         hasExplicitCodecTypes = AVIF_FALSE;
     }
 
-    uint32_t smallDimensionsFlag = image->width <= (1 << 7) && image->height <= (1 << 7);
-    const uint32_t codecConfigSize = 4; // 'av1C' always uses 4 bytes.
+    // _minus1 is encoded for these fields.
+    AVIF_ASSERT_OR_RETURN(image->width != 0 && image->height != 0);
```

Comment:
Since `image` comes from user input, if `image->width` or `image->height` is 0, it is an invalid input argument rather than an internal error. So `AVIF_RESULT_INVALID_ARGUMENT` seems more appropriate than `AVIF_RESULT_INTERNAL_ERROR` unless we expect libavif to have validated `image->width` and `image->height` before reaching here.

Reply:
I think it is enforced by

https://github.com/AOMediaCodec/libavif/blob/c6807414018046e280ed1f5f24d2ce1a04b16ca3/src/write.c#L1719

and

https://github.com/AOMediaCodec/libavif/blob/c6807414018046e280ed1f5f24d2ce1a04b16ca3/src/write.c#L3159

Reply:
Thank you for the analysis on `colorData->size`. To me an issue with a codec is still an internal libavif error because libavif dependencies are the responsibility of libavif, unless we allow user-defined codecs such as in https://github.com/AOMediaCodec/libavif/pull/2381.

If `AVIF_RESULT_INTERNAL_ERROR` is improper, I would suggest returning `avifGetErrorForItemCategory()` instead, as is done for other codec errors. What do you think?

Reply:
> a malfunctioning codec could call `avifCodecEncodeOutputAddSample()` with a zero-sized sample

Should this be caught somewhere? Can a zero-sized sample be valid in regular HEIF?